logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueReader.java

```diff
@@ -271,6 +320,20 @@ public void setCurrentReaderAndPosition(Path segmentPath, long position) throws
                 .ifPresent(reader -> currentReader = reader);
     }
 
+    private void removeSegmentsBefore(Path validSegment) throws IOException {
+        Files.list(queuePath)
+                .filter(p -> p.compareTo(validSegment) < 0)
```

Comment:
It seems segments are given name like `1.log`, `2.log`, etc... and here we are lexicographically comparing. Any case which `2.log` will be created before `1.log` segment and `2.log` needs to be removed first? If segment sequence reaches `Integer.MAX_VALUE`, it may possible. So, my point is if we have such case, better to use timestamp for comparison.

Reply:
Your concern is valid, and that limit correspond about to 20 PB of data processed by DLQ. Also we can't exclude that 2 segments has the same timestamp, so the best solution I can foresee is to use timestamp + filename ordering criteria.
WDYT?

Reply:
This problem, could eventually happens also during DLQ reader [creation](https://github.com/elastic/logstash/blob/04f3e32ffa60fe61ba8cc7f7741ab95df906d11a/logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueReader.java#L80).

Reply:
Thanks @andsel for details.
`timestamp + filename` sounds good to me for this case.